pkg/ruler/compat_test.go

```diff
@@ -241,3 +241,21 @@ func TestMetricsQueryFuncErrors(t *testing.T) {
 		})
 	}
 }
+
+func TestMetricsQueryFuncMetrics(t *testing.T) {
```

Comment:
`TestMetricsQueryFuncErrors` already tests `MetricsQueryFunc`.

I would expect new test to test only `RecordAndReportRuleQueryMetrics`.